ui/core/src/utils/text.ts

```diff
@@ -34,9 +34,9 @@ export function getVariableDisplayName(variable: Variable) {
 /**
  * If the variable has a display name, return the datasource display name
  * Else, only return the datasource name
- * @param variable Project or Global datasource
+ * @param datasource Project or Global datasource
  */
-export function getDatasourceDisplayName(datasource: Datasource) {
+export function getDatasourceDisplayName(datasource: ProjectDatasource | GlobalDatasource) {
```

Comment:
You should reuse the `Datasource` type you created here

`export type Datasource = ProjectDatasource | GlobalDatasource;`